### PR TITLE
Connection status event #12

### PR DESCRIPTION
**Pull request by yozik04**

### CODE REVIEW

**Comment by yozik04** [comment]
@elupus take a look. Will you be happy with it?

File: nibe/connection/nibegw.py [hunk 1]
```diff
@@ -77,6 +82,7 @@ async def start(self):
         )
 
     def connection_made(self, transport):
+        self.notify_event_listeners(self.CONNECTION_STATUS_EVENT, status="connected")
```
Comment (elupus) [inline nibe/connection/nibegw.py:85]: I think this callback is called as soon as the udp socket starts listening. So we dont know we have received anything. I think we could notify on product data too?

Comment (yozik04) [reply nibe/connection/nibegw.py:85]: Only if you request product data =)

**Comment by yozik04** [comment]
Now it should be OK.

File: nibe/connection/nibegw.py [hunk 2]
```diff
@@ -63,24 +76,29 @@ def __init__(
...
             proto=socket.IPPROTO_UDP,
         )
 
     def connection_made(self, transport):
+        self._set_status(ConnectionStatus.LISTENING)
         self._transport = transport
 
     def datagram_received(self, data, addr):
         logger.debug(f"Received {hexlify(data)} from {addr}")
+        self._set_status(ConnectionStatus.CONNECTED)
```
Comment (elupus) [inline nibe/connection/nibegw.py:101]: One note here. I noticed my chromecast audio sends udp data on port 9999, which is default for nibegw. Its should likely check for the addr here to verify it's from the expected remote address.

**Review by elupus — APPROVED** [review]
Looks good in general

File: nibe/connection/nibegw.py (hunk 2 above)
Comment (yozik04) [reply nibe/connection/nibegw.py:101]: We did try to add that filter before. But unfortunately when run in docker you do not get real IP address in UDP packets.

**Pull request merged**